GRASS community map

lucadelu opened the pull request:
Related to #164

lucadelu committed:
started to work on community map

lucadelu committed:
added missing geojson folder

lucadelu committed:
fixed geojson sintax

lucadelu committed:
added institutions contributing to GRASS into map

lucadelu committed:
improved popup info

lucadelu committed:
changed logo into image to keep the same code in the page

lucadelu committed:
improved input style

lucadelu committed:
renamed geojson file

lucadelu committed:
improved README file for geojson

lucadelu committed:
fixed name

lucadelu committed:
Merge branch 'master' into communitymap

lucadelu committed:
fixed typos and updates to newer hugo version

lucadelu committed:
several small improvements

lucadelu committed:
improved english

veroandreo commented on content/about/map.md:
```diff
@@ -0,0 +1,9 @@
+
+### Add yourself
+
+To add yourself to the GRASS user map you need to follow [these instructions](https://github.com/OSGeo/grass-website/static/geojson/README.md).
```
```suggestion
To add yourself to the GRASS community map, follow [these instructions](https://github.com/OSGeo/grass-website/static/geojson/README.md).
```

veroandreo commented on static/geojson/README.md:
```diff
@@ -0,0 +1,69 @@
+# GRASS GIS map
```
```suggestion
# GRASS community map
```

veroandreo commented on static/geojson/README.md:
```diff
@@ -0,0 +1,69 @@
+# GRASS GIS map
+
+In this folder you can find geographical data related to GRASS GIS community. The data are provided in GeoJSON format.
```
```suggestion
In this folder you can find geographical data related to the GRASS community. The data are provided in GeoJSON format.
```

veroandreo commented on static/geojson/README.md:
```diff
@@ -0,0 +1,69 @@
+
+## User map
+
+The [user.geojson file](https://github.com/OSGeo/grass-website/static/geojson/user.geojson) contain the position of GRASS GIS users.
```
```suggestion
The [user.geojson file](https://github.com/OSGeo/grass-website/static/geojson/user.geojson) contains the positions of GRASS users and community members who willingly added themselves.
```

veroandreo commented on static/geojson/README.md:
```diff
@@ -0,0 +1,69 @@
+
+The [user.geojson file](https://github.com/OSGeo/grass-website/static/geojson/user.geojson) contain the position of GRASS GIS users.
+The [provider.geojson file](https://github.com/OSGeo/grass-website/static/geojson/provider.geojson) contain the position of any companies,
+research center, university providing activities (specially development and courses) related to GRASS GIS.
```
```suggestion
The [provider.geojson file](https://github.com/OSGeo/grass-website/static/geojson/provider.geojson) contains the position of companies,
research centers and universities offering support related to GRASS, i.e., commercial support, new features development and implementations, courses, etc.
```

veroandreo commented on static/geojson/README.md:
```diff
@@ -0,0 +1,69 @@
+The [provider.geojson file](https://github.com/OSGeo/grass-website/static/geojson/provider.geojson) contain the position of any companies,
+research center, university providing activities (specially development and courses) related to GRASS GIS.
+
+## Add new record
```
```suggestion
## Add new records to the map
```

veroandreo commented on static/geojson/README.md:
```diff
@@ -0,0 +1,69 @@
+
+## Add new record
+
+If you want to add a new record to one of the GeoJSON files you have to ways
```
```suggestion
```

veroandreo commented on static/geojson/README.md:
```diff
@@ -0,0 +1,69 @@
+
+### User without Git/Github experience
+
+* You need to create a [Github account](https://github.com/join)
```
```suggestion
* Create a [Github account](https://github.com/join)
```

veroandreo commented on static/geojson/README.md:
```diff
@@ -0,0 +1,69 @@
+
+* You need to create a [Github account](https://github.com/join)
+* Sign in and move to https://github.com/OSGeo/grass-website/ repository
+* Choose the file to modify into [GeoJSON files](https://github.com/OSGeo/grass-website/static/geojson/)
```
```suggestion
* Choose which file to modify from the [GeoJSON files](https://github.com/OSGeo/grass-website/static/geojson/)
```

veroandreo commented on static/geojson/README.md:
```diff
@@ -0,0 +1,69 @@
+* Choose the file to modify into [GeoJSON files](https://github.com/OSGeo/grass-website/static/geojson/)
+* Click the pencil button to modify the file
+* Add a new feature copying the feature template in the bottom of the page 
+* When you save the file, select `Create a new branch for this commit and start a pull request.`
```
```suggestion
* When saving the file, select `Create a new branch for this commit and start a pull request.`
```

neteler commented on content/about/map.md:
```diff
@@ -0,0 +1,9 @@
+---
+title: "GRASS community map"
+date: 2018-12-29T11:02:05+06:00
```
```suggestion
date: 2026-07-14T11:02:05+06:00
```

neteler commented on static/geojson/provider.geojson:
```diff
@@ -0,0 +1,21 @@
+    {
+      "type": "Feature",
+      "properties": {
+        "name": "Mundialis",
```
```suggestion
        "name": "mundialis",
```

neteler committed:
minor typo

veroandreo reviewed (changes requested):
Nice!! Thanks @lucadelu! I suggested some wording changes here and there. 
I wonder if it is possible to add some white space between the map and the footer. Right now, there's none.

<img width="1187" height="261" alt="Image" src="https://github.com/user-attachments/assets/75142ecb-442d-449f-8672-2c0b7836b34e" />

lucadelu committed:
Apply suggestions from code review

Thanks to @veroandreo @neteler

Co-authored-by: Veronica Andreo <veroandreo@gmail.com>
Co-authored-by: Markus Neteler <markus@neteler.org>

lucadelu committed:
add space under the map

lucadelu committed:
Merge remote-tracking branch 'upstream/master' into communitymap

lucadelu committed:
fixed popup removing when change layer

veroandreo reviewed (dismissed):
I tested the latest change and now the behavior is right; if I display users and then switch to providers, the active user is closed. Thanks Luca!

lucadelu committed:
merged master branch

cwhite911 reviewed (changes requested):
@lucadelu like we talked about the repo  now manages all third-party frontend assets (Bootstrap, Font Awesome, etc..) through package.json plus [[module.mounts]] in `config.toml:26-73`, with Renovate handling upgrades.

You need to:

## Adding openlayers as a dependency
- Add ol to package.json dependencies and mount node_modules/ol/dist/ol.js and node_modules/ol/ol.css the same way the other packages are mounted.
- Delete themes/grass/static/plugins/ol-6.3.1/ entirely, especially the 3.6 MB ol.js.map, which permanently bloats repo history.

## OpenLayers API changes
Two API changes when moving from 6.3.1 (April 2020) to current OL (ol v10.9.0): `ol.control.defaults()` becomes `ol.control.defaults.defaults()` (map.js:86), and `autoPanAnimation` (map.js:14) is removed in favor of `autoPan: { animation: { duration: 250 } }`.

## Styling
No CSS for the map or popup exists anywhere. There are no `.map` height rules and no `.ol-popup / .ol-popup-closer` styles in themes/grass/assets/sass/. Even with ol.js loaded, `#map` renders `0 px` tall and the popup is unstyled. Needs SCSS added to the theme (use the brand tokens) plus the mounted ol.css.

Addendum: wiring the map SCSS

The theme compiles one bundle from themes/grass/assets/sass/main.scss via toCSS in head.html:40-50, with $gs-* tokens in _colors.scss and custom rules in _styles.scss imported last so they win the cascade. The map styles should follow that pattern, in three parts.

1. OL's own stylesheet, through the Sass pipeline

Once ol is an npm dependency, don't mount ol.css as a separate <link>. Add node_modules to the Sass load paths in themes/grass/layouts/partials/head.html:

```go
go-html-template
{{ $cssopts := dict
  "targetPath" "css/style.css"
  "outputStyle" "compressed"
  "enableSourceMap" false
  "includePaths" (slice "node_modules/bootstrap/scss" "node_modules")
}}
```

Then Dart Sass inlines OL's plain CSS when imported without extension (a .css file imported extensionless is inlined, not left as a URL import). OL's control/viewport CSS is only a few KB compressed, so shipping it in the global bundle is cheaper than a second render-blocking request on one page.

2. New partial themes/grass/assets/sass/_map.scss

```scss
// Community map (/about/map). OL base styles come from the ol npm package;
// everything below rebrands controls and the popup to the GRASS tokens.
@import "ol/ol";

.community-map {
  height: 70vh;
  min-height: 420px;
  border: 1px solid $gs-grey-alt-color;
  border-radius: 0.25rem; // modest geometry, match existing cards
  overflow: hidden;       // clip tiles to the radius

  @include media-breakpoint-down(md) {
    height: 55vh;
  }
}

// Zoom / attribution controls: blue for structure, per brand
.ol-control button {
  background-color: $gs-secondary-alt-color;

  &:hover,
  &:focus {
    background-color: $gs-secondary-light-color;
  }
}

// Popup (OL overlay example markup, branded)
.ol-popup {
  position: absolute;
  bottom: 12px;
  left: -50px;
  min-width: 200px;
  padding: 15px;
  background-color: $gs-white-base-color;
  border: 1px solid $gs-grey-alt-color;
  border-radius: 0.25rem;
  box-shadow: 0 5px 15px $gs-black-color--lightest; // matches card shadow in _styles.scss

  // anchor arrow
  &::after,
  &::before {
    content: "";
    position: absolute;
    top: 100%;
    width: 0;
    height: 0;
    border: solid transparent;
    pointer-events: none;
  }

  &::after {
    left: 48px;
    margin-left: -10px;
    border-width: 10px;
    border-top-color: $gs-white-base-color;
  }

  &::before {
    left: 48px;
    margin-left: -11px;
    border-width: 11px;
    border-top-color: $gs-grey-alt-color;
  }
}

.ol-popup-closer {
  position: absolute;
  top: 2px;
  right: 8px;
  color: $gs-grey-color;
  text-decoration: none;

  &::after {
    content: "\2716";
  }

  &:hover {
    color: $gs-primary-dark-color;
  }
}
```

Notes on the choices:

- @include media-breakpoint-down works because main.scss imports Bootstrap in the shared @import scope before this partial.
- Blue controls, green data: the brand rule is green for interactive data and blue for structural chrome. The marker styles in map.js already hardcode #088B36, which is exactly $gs-primary-dark-color, so points read as GRASS green against blue controls. Ask the author to add a comment in map.js noting those hex values mirror the SCSS tokens, since JS can't read Sass variables.
- Shadow and radius values copy the existing card treatment in _styles.scss rather than inventing new elevation.

3. Register the partial and fix the markup hook

In main.scss, after the existing imports (custom rules must stay after bootstrap):

```scss
@import "styles";
@import "map";
```

And in `themes/grass/layouts/about/map.html`, replace `class="map pb-4"` with `class="community-map mb-4"`: `.map` is too generic a class to put in the global bundle, and pd-5 / pb-4 on the map div gets replaced by the explicit height anyway.


## Bugs
- XSS surface in the popup. Build the popup with `createElement/textContent/setAttribute` and validate that url and image are http(s).
- Stale highlight: clicking feature B while A is highlighted overwrites oldfeat (map.js:105) without resetting A's style, so A stays enlarged. Call `remove_popup()` at the top of the click handler.
- feat at map.js:98 is an implicit global (no declaration).
- `themes/grass/layouts/about/map.html`: the .container div is never closed (only .row is, before </section>).
- {{ if eq .RelPermalink "/about/map/" }} in footer.html is fragile: it breaks under a subpath baseURL and for any future translated page (/it/about/map/). {{ if eq .Layout "map" }} is robust and simpler.
- map.js is served minified but without fingerprint/SRI, unlike every other bundle in footer.html:110-112. Pipe it through resources.Fingerprint "sha512" for consistency and cache busting.
- Layer switching (map.js:129-138) rebuilds a whole new source; vsource.setUrl(...) + vsource.refresh() suffices, and the trailing .changed() is redundant. Also the layer title stays "Users" after switching to providers.

## pre-commit
Make sure to use `.pre-commit` now it will fix a lot of formatting bugs.

lucadelu committed:
Apply suggestions from code review

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

lucadelu committed:
updated community map to use new implementation

lucadelu committed:
fixed  MD013/line-length

lucadelu committed:
fixed openlayers style

lucadelu committed:
small improvements

neteler commented on static/geojson/user.geojson:
```diff
@@ -0,0 +1,20 @@
+      "type": "Feature",
+      "properties": {
+        "name": "Luca Delucchi",
+        "url": "http://www.lucadelu.org",
```
```suggestion
        "url": "https://www.lucadelu.org",
```

lucadelu committed:
Update static/geojson/user.geojson

Co-authored-by: Markus Neteler <markus@neteler.org>

cwhite911 commented on themes/grass/assets/sass/_community_map.scss:
Let's rename this file to community_map, and update the #map id to #community-map, to remove ambiguity about what it's referencing.

lucadelu committed:
renamed files and fixed problem in markdown

cwhite911 commented on themes/grass/assets/js/community_map.js:
The initial centering was off.
`useGeographic()` sets the user projection to lon/lat, but the center is still passed through ol.proj.transform(...) to Web Mercator, so the map opens near the North Pole instead of [0, 5] ). This single 


* rewrite fixes the center
* converts to ESM import so OL is tree-shaken instead of shipping the full 898 KB build.
* applies const/let (the top-level vars currently leak onto window), fixes the stale-highlight bug, and removes the innerHTML XSS by building the popup with DOM APIs and validating URLs.

Replace the entire contents of `themes/grass/assets/js/community_map.js` with:

```js
// Community map (/about/map). Bundled by Hugo's js.Build (esbuild), which
// tree-shakes OpenLayers to only the modules imported below.
import Map from 'ol/Map.js';
import View from 'ol/View.js';
import Overlay from 'ol/Overlay.js';
import TileLayer from 'ol/layer/Tile.js';
import VectorLayer from 'ol/layer/Vector.js';
import OSM from 'ol/source/OSM.js';
import VectorSource from 'ol/source/Vector.js';
import GeoJSON from 'ol/format/GeoJSON.js';
import Style from 'ol/style/Style.js';
import CircleStyle from 'ol/style/Circle.js';
import Fill from 'ol/style/Fill.js';
import Stroke from 'ol/style/Stroke.js';
import { useGeographic } from 'ol/proj.js';
import { defaults as defaultControls } from 'ol/control/defaults.js';

const container = document.getElementById('popup');
const content = document.getElementById('popup-content');
const closer = document.getElementById('popup-closer');

if (container && content && closer) {
  // Work in [lon, lat] so raw GeoJSON coordinates render without reprojection.
  useGeographic();

  let oldfeat = null;
  let vecturl = '/geojson/user.geojson';

  // #088B36 mirrors $gs-primary-dark-color in _map.scss (JS can't read Sass tokens).
  const markerStyle = (radius, width) =>
    new Style({
      image: new CircleStyle({
        radius: radius,
        fill: new Fill({ color: 'rgb(255, 255, 255)' }),
        stroke: new Stroke({ color: '#088B36', width: width }),
      }),
    });
  const defaultStyle = markerStyle(4, 2);
  const highlightStyle = markerStyle(8, 3);

  const overlay = new Overlay({
    element: container,
    autoPan: { animation: { duration: 250 } },
  });

  function removePopup() {
    if (oldfeat) {
      oldfeat.setStyle(defaultStyle);
      oldfeat = null;
    }
    overlay.setPosition(undefined);
    closer.blur();
  }

  closer.onclick = function () {
    removePopup();
    return false;
  };

  const vectorLayer = new VectorLayer({
    source: new VectorSource({ url: vecturl, format: new GeoJSON() }),
    style: defaultStyle,
  });

  const map = new Map({
    target: 'map',
    layers: [new TileLayer({ source: new OSM() }), vectorLayer],
    overlays: [overlay],
    controls: defaultControls({ attributionOptions: { collapsible: false } }),
    view: new View({ center: [0, 5], zoom: 2 }),
  });

  // Feature properties come from community-edited GeoJSON: never inject as HTML.
  function isHttpUrl(value) {
    if (!value) return false;
    try {
      const parsed = new URL(value, window.location.origin);
      return parsed.protocol === 'http:' || parsed.protocol === 'https:';
    } catch (e) {
      return false;
    }
  }

  function renderPopup(feature) {
    content.replaceChildren();
    const name = feature.get('name') || '';
    const url = feature.get('url');
    const image = feature.get('image');
    const description = feature.get('description');

    const line = document.createElement('p');
    if (isHttpUrl(image)) {
      const img = document.createElement('img');
      img.src = image;
      img.width = 75;
      img.alt = name;
      line.appendChild(img);
    }
    if (isHttpUrl(url)) {
      const link = document.createElement('a');
      link.href = url;
      link.target = '_blank';
      link.rel = 'noopener noreferrer';
      link.textContent = name;
      line.appendChild(link);
    } else {
      line.appendChild(document.createTextNode(name));
    }
    content.appendChild(line);

    if (description) {
      const desc = document.createElement('p');
      const small = document.createElement('small');
      small.textContent = description;
      desc.appendChild(small);
      content.appendChild(desc);
    }
  }

  map.on('click', function (evt) {
    removePopup(); // reset any previously highlighted feature first
    const feature = map.forEachFeatureAtPixel(evt.pixel, (f) => f);
    if (!feature) return;
    feature.setStyle(highlightStyle);
    oldfeat = feature;
    renderPopup(feature);
    overlay.setPosition(evt.coordinate);
  });

  document.querySelectorAll("button[name='layers']").forEach(function (button) {
    button.addEventListener('click', function () {
      removePopup();
      vecturl = '/geojson/' + button.value + '.geojson';
      vectorLayer.setSource(
        new VectorSource({ url: vecturl, format: new GeoJSON() })
      );
    });
  });
}
```

cwhite911 replied:
Replace what is in `community_map.js` with this.

cwhite911 commented on themes/grass/layouts/partials/footer.html:
```diff
@@ -108,5 +111,14 @@ <h6 class="text-uppercase text-light">Contribute</h6>
+<script src="{{ $oljs.Permalink }}"></script>
+{{ $mapscript := resources.Get "js/community_map.js" }}
+<script src="{{ $mapscript.Permalink }}"></script>
+{{ end }}
```
js.Build's default IIFE output stays a plain <script>; esbuild resolves ol from the mounted node_modules; es2020 covers OL 10's syntax. This also gives the bundle SRI + fingerprinting like the other scripts, and .Layout is robust where the RelPermalink string match was not.

```suggestion
{{ if eq .Layout "map" }}
{{ "<!-- Map Script -->" | safeHTML }}
{{ $map := resources.Get "js/map.js" | js.Build (dict "minify" true "target" "es2020") | fingerprint }}
<script src="{{ $map.Permalink }}" integrity="{{ $map.Data.Integrity }}"></script>
{{ end }}
```

cwhite911 commented on themes/grass/layouts/partials/footer.html:
```diff
@@ -108,5 +111,14 @@ <h6 class="text-uppercase text-light">Contribute</h6>
 </body>
-</html>
+
+</html>
```
Add new line at end of file

cwhite911 commented on themes/grass/layouts/partials/head.html:
```diff
@@ -33,23 +33,30 @@
   <link rel="stylesheet" href="{{ $highlightjs_styles.Permalink }}">
   {{ $flagicons := resources.Get "css/flag-icons/flag-icons.min.css" }}
   <link rel="stylesheet" href="{{ $flagicons.Permalink }}">
+  {{ if or (eq .RelPermalink "/about/map/") (eq .RelPermalink "/about/map") }}
```
```suggestion
   {{ if eq .Layout "community-map" }}
```

cwhite911 commented on themes/grass/layouts/partials/head.html:
```diff
@@ -33,23 +33,30 @@
   {{ $bootstrap_js := resources.Get "js/bootstrap/bootstrap.bundle.min.js" | fingerprint }}
   <script src="{{ $bootstrap_js.Permalink }}" integrity="{{ $bootstrap_js.Data.Integrity }}"></script>
+  {{ if eq .RelPermalink "/about/map/" }}
+  {{ end }}
```
Remove unused block
```suggestion
```

cwhite911 commented on themes/grass/layouts/partials/head.html:
```diff
@@ -74,4 +81,4 @@
 
 <body>
-  <noscript>Please enable javascript :)</noscript>
+  <noscript>Please enable javascript :)</noscript>
```
Add new line at end of file

cwhite911 commented on config.toml:
```diff
@@ -72,6 +72,14 @@ priority = 0.5
     source = "node_modules/photoswipe/dist/photoswipe-lightbox.esm.js"
     target = "assets/js/photoswipe/photoswipe-lightbox.esm.js"
 
+    [[module.mounts]]
+    source = "node_modules/ol/dist/ol.js"
+    target = "assets/js/ol/ol.js"
```
Delete the JS mount, since js.Build imports OL directly.

```suggestion
```

cwhite911 commented on themes/grass/layouts/about/community-map.html:
```diff
@@ -0,0 +1,47 @@
+    </div>
+</section>
+
+{{ partial "footer.html" . }}
```
Add new line at the end of the file.

cwhite911 commented on themes/grass/layouts/about/community-map.html:
```diff
@@ -0,0 +1,47 @@
+        <h3>Choose what to see</h3>
+        <button type="button" name='layers' value='user' class="btn btn-primary pd-5">Users</button>
+        <button type="button" name='layers' value='provider' class="btn btn-primary pd-5">Providers</button>
+        {{ .Content }}
+      </div>
+    </div>
+</section>
```
Add missing closing div
```suggestion
      </div>
    </div>
  </div>
</section>
```

cwhite911 commented on static/geojson/README.md:
```diff
@@ -0,0 +1,73 @@
+
+## User map
+
+The [user.geojson file](https://github.com/OSGeo/grass-website/static/geojson/user.geojson)
```
```suggestion
The [user.geojson file]((https://github.com/OSGeo/grass-website/blob/master/static/geojson/user.geojson)
```

cwhite911 commented on static/geojson/README.md:
```diff
@@ -0,0 +1,73 @@
+The [user.geojson file](https://github.com/OSGeo/grass-website/static/geojson/user.geojson)
+contains the positions of GRASS users and community members who willingly
+added themselves.
+The [provider.geojson file](https://github.com/OSGeo/grass-website/static/geojson/provider.geojson)
```
```suggestion
The [provider.geojson file](https://github.com/OSGeo/grass-website/blob/master/static/geojson/provider.geojson)
```

cwhite911 commented on static/geojson/README.md:
```diff
@@ -0,0 +1,73 @@
+
+* Create a [Github account](https://github.com/join)
+* Sign in and move to <https://github.com/OSGeo/grass-website/> repository
+* Choose which file to modify from the [GeoJSON files](https://github.com/OSGeo/grass-website/static/geojson/)
```
```suggestion
* Choose which file to modify from the [GeoJSON files](https://github.com/OSGeo/grass-website/tree/master/static/geojson/)
```

cwhite911 commented on content/about/map.md:
```diff
@@ -0,0 +1,9 @@
+
+### Add yourself
+
+To add yourself to the GRASS community map, follow [these instructions](https://github.com/OSGeo/grass-website/static/geojson/README.md).
```
```suggestion
To add yourself to the GRASS community map, follow [these instructions](https://github.com/OSGeo/grass-website/blob/master/static/geojson/README.md).
```

cwhite911 reviewed (changes requested):
Check-json only matches *.json, so a malformed community edit fails silently (the map just renders nothing). Please add geojson coverage in .pre-commit-config.yaml:

      - id: check-json
        files: \.(json|geojson)$
        types: [text]

lucadelu committed:
Apply suggestions from code review

Co-authored-by: Corey White <smortopahri@gmail.com>

lucadelu committed:
fixed loading js file

lucadelu committed:
fixed missing line at the end of files

cwhite911 committed:
Merge branch 'master' into communitymap

cwhite911 commented on themes/grass/assets/sass/README.md:
```diff
@@ -13,4 +16,6 @@ Hugo Integration Documentation: [https://gohugo.io/functions/css/sass/#dart-sass
 - `assets/sass/_fonts.scss`: Contains font variables used throughout the theme.
 - `assets/sass/_variables.scss`: Contains additional Sass variables for customization.
-- `layouts/partials/head.html`: The HTML partial that includes the compiled CSS files in the site head.
+- `assets/sass/_map.scss`: Contain additional Sass variable for community map
```
Rename _community_map.scss

cwhite911 commented on content/about/community-map.md:
The file needs to be renamed to `community-map.md`

cwhite911 commented on themes/grass/assets/sass/README.md:
```diff
@@ -13,4 +16,6 @@ Hugo Integration Documentation: [https://gohugo.io/functions/css/sass/#dart-sass
 - `assets/sass/_fonts.scss`: Contains font variables used throughout the theme.
 - `assets/sass/_variables.scss`: Contains additional Sass variables for customization.
-- `layouts/partials/head.html`: The HTML partial that includes the compiled CSS files in the site head.
+- `assets/sass/_map.scss`: Contain additional Sass variable for community map
```
Update the docs to match the new name.

lucadelu replied:
Done

cwhite911 commented on themes/grass/layouts/about/community-map.html:
```diff
@@ -0,0 +1,49 @@
+  <div class="container">
+    <div class="row">
+      <div class="col-lg-10">
+        <div class="map mb-3" id="community-map"></div>
```
```suggestion
        <div class="mb-3" id="community-map"></div>
```

cwhite911 committed:
Fix community map: load OpenLayers as UMD, not an esbuild bundle

Hugo's js.Build (esbuild) breaks OpenLayers' internal instanceof checks
(View, Style), so the map crashed at construction with
'TypeError: view.then is not a function' and a bogus 'No fill, stroke,
point, or text symbolizer' error. Load ol as its standalone UMD build
(window.ol) via a script tag instead of bundling it.

Also: fix the initial view center under useGeographic() (plain [lon, lat],
no proj transform), build the popup with DOM APIs instead of innerHTML,
validate feature URLs are http(s), reset the previously highlighted
feature on each click, and switch var to let/const.

cwhite911 committed:
Fix community map docs and lint nits

- Fix malformed user.geojson link in README (stray double paren)
- Add missing EOF newlines to main.scss and _community_map.scss

lucadelu committed:
fixed name of scss file

cwhite911 commented on themes/grass/assets/sass/_community_map.scss:
```diff
@@ -0,0 +1,85 @@
+// Community map (/about/map). OL base styles are loaded separately from the
```
```suggestion
// Community map (/about/community-map). OL base styles are loaded separately from the
```

cwhite911 committed:
Apply suggestions from code review

Co-authored-by: Corey White <smortopahri@gmail.com>

cwhite911 reviewed (commented):
@lucadelu will you add this too.

Check-json only matches *.json, so a malformed community edit fails silently (the map just renders nothing). Please add geojson coverage in .pre-commit-config.yaml:

```yml
      - id: check-json
        files: \.(json|geojson)$
        types: [text]
```

lucadelu committed:
added records to geojson

lucadelu committed:
added check for geojson files

lucadelu committed:
added check for geojson files

cwhite911 reviewed (approved):
Looks great!
